integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -93,6 +93,438 @@ describe.skipIf(environment.missingServer, "MongoDB Client", function () {
   });
 
   describe("MongoDBCollection", function () {
+    const insertedId1 = 1;
+    const insertedId2 = 2;
+    const insertedId3 = 3;
+    const insertedText = "Test document";
+    const nonExistentId = 100;
+
+    async function insertThreeDocuments(): Promise<void> {
+      const { insertedIds } = await collection.insertMany([
+        { _id: insertedId1, text: insertedText },
+        { _id: insertedId2, text: insertedText },
+        { _id: insertedId3, text: insertedText },
+      ]);
+      expect(insertedIds).to.have.length(3);
+    }
+
+    describe("#find", function () {
+      it("returns all documents using empty filter", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const docs = await collection.find();
+        expect(docs).to.have.length(3);
+        for (const doc of docs) {
+          expect(doc).to.have.all.keys("_id", "text");
+        }
+      });
+
+      it("returns all documents excluding a field using 'projection' option", async function (this: AppContext &
+        Mocha.Context) {
+        await insertThreeDocuments();
+
+        const docs = await collection.find({}, { projection: { text: false } });
+        expect(docs).to.have.length(3);
+        for (const doc of docs) {
+          expect(doc).to.have.property("_id");
+          expect(doc).to.not.have.property("text");
+        }
+      });
+
+      it("returns documents using query selector", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const docs = await collection.find({ _id: { $gt: insertedId1 } });
+        expect(docs).to.have.length(2);
+        for (const doc of docs) {
+          expect(doc._id > insertedId1).to.be.true;
+        }
+      });
+
+      it("returns empty array if collection is empty", async function (this: AppContext & Mocha.Context) {
+        const docs = await collection.find();
```

Comment:
You will probably want to move this to a separate describe if you do go down the `beforeEach` route, rest seem to use it.